src/mame/snk/bbusters.cpp

```diff
@@ -313,20 +312,20 @@ uint32_t bbusters_state::screen_update(screen_device &screen, bitmap_ind16 &bitm
 
 /*******************************************************************************/
 
-void bbusters_state::bbusters_map(address_map &map)
+void bbusters_state::main_map(address_map &map)
 {
 	map(0x000000, 0x07ffff).rom();
-	map(0x080000, 0x08ffff).ram().share("ram");
-	map(0x090000, 0x090fff).ram().w(FUNC(bbusters_state::video_w)).share("tx_videoram");
+	map(0x080000, 0x08ffff).ram();
+	map(0x090000, 0x090fff).ram().w(FUNC(bbusters_state::tx_vram_w)).share(m_tx_vram);
 	map(0x0a0000, 0x0a0fff).ram().share("spriteram1");
 	map(0x0a1000, 0x0a7fff).ram();     /* service mode */
 	map(0x0a8000, 0x0a8fff).ram().share("spriteram2");
 	map(0x0a9000, 0x0affff).ram();     /* service mode */
-	map(0x0b0000, 0x0b1fff).ram().w(FUNC(bbusters_state::pf_w<0>)).share("pf1_data");
-	map(0x0b2000, 0x0b3fff).ram().w(FUNC(bbusters_state::pf_w<1>)).share("pf2_data");
+	map(0x0b0000, 0x0b1fff).ram().w(FUNC(bbusters_state::pf_vram_w<0>)).share(m_pf_vram[0]);
+	map(0x0b2000, 0x0b3fff).ram().w(FUNC(bbusters_state::pf_vram_w<1>)).share(m_pf_vram[1]);
 	map(0x0b4000, 0x0b5fff).ram();     /* service mode */
-	map(0x0b8000, 0x0b8003).writeonly().share("pf1_scroll_data");
-	map(0x0b8008, 0x0b800b).writeonly().share("pf2_scroll_data");
+	map(0x0b8000, 0x0b8003).writeonly().share(m_pf_scroll_ram[0]);
+	map(0x0b8008, 0x0b800b).writeonly().share(m_pf_scroll_ram[1]);
```

Comment:
It isn't RAM out of being set as `writeonly()`, more like `m_pf_scroll_reg[N]`.

I don't know if this driver has been written with schematics at hand/has been tested on real HW, but regardless even if readable the rule of thumb for scroll areas is `reg` if it touches layer global/local positions only, `[v]ram` if it's a table of some sort.